src/http_server.cc

```diff
@@ -4487,16 +4546,38 @@ HTTPAPIServer::Create(
     const std::shared_ptr<SharedMemoryManager>& shm_manager, const int32_t port,
     const bool reuse_port, const std::string& address,
     const std::string& header_forward_pattern, const int thread_cnt,
+    const RestrictedFeatureMap& restricted_features,
     std::unique_ptr<HTTPServer>* http_server)
 {
   http_server->reset(new HTTPAPIServer(
       server, trace_manager, shm_manager, port, reuse_port, address,
-      header_forward_pattern, thread_cnt));
+      header_forward_pattern, thread_cnt, restricted_features));
 
   const std::string addr = address + ":" + std::to_string(port);
   LOG_INFO << "Started HTTPService at " << addr;
 
   return nullptr;
 }
 
+bool
+HTTPAPIServer::RestrictedAPI(evhtp_request_t* req, const std::string api)
+{
+  auto restricted_api = restricted_apis_.find(api);
+  if (restricted_api == restricted_apis_.end()) {
+    return false;
+  }
```

Comment:
There are a couple things I think we could do:

1) Have the table be a fixed array keyed on an enum  (the number of restricted apis is fixed and we don't need a map)
2) introduce a thread safe function level static to hold the value:

```
static const auto restricted_api_key_value = ..
  
if (restricted_api_key_value && Restricted()) {
  return;
}
```

Given the number of checks already in the hotpath - let me know what you think - I'm not opposed to it - but not sure if it would be meaningful.